src/components/CreateOrUpdateEvent.svelte

```diff
@@ -0,0 +1,176 @@
+<script>
+  import LocationInput from 'components/LocationInput.svelte'
+  import Uploader from 'components/Uploader.svelte'
+  import { throwError } from 'data/error.js'
+  import { events, update, create } from 'data/events.js'
+  import { me } from 'data/user.js'
+
+  import { format, addDays } from 'date-fns'
+  import { goto, params } from '@roxi/routify'
+
+  const tomorrow = format(addDays(Date.now(), 1), 'yyyy-MM-dd')
+
+  $: event = {}
+  $: existingEvent = $events.find(({ id }) => id === $params.eventId)
+  $: initializeUpdates(existingEvent)
+  $: isNew = !event.id
+
+  new Date().getDate
+
+  const parseDate = dateString => {
+    let date = new Date(dateString)
+    let month = date.getMonth() + 1
+    let dayOfMonth = date.getDate() + 1
+    let monthLength = `${month}`.length
+    let dateLength = `${dayOfMonth}`.length
+    return `${date.getFullYear()}-${monthLength == 1 ? `0${month}` : month}-${dateLength == 1 ? `0${dayOfMonth}` : dayOfMonth}`
+  }
+
+  function initializeUpdates(eventBeingEdited) {
+    if (!eventBeingEdited) return
+    event = Object.assign({}, eventBeingEdited)
+    event.start_date = parseDate(event.start_date)
+    event.end_date = parseDate(event.end_date)
+  }
+
+  function assertHas(value, errorMessage) {
+    if (!value) {
+      throwError(errorMessage)
+    }
+  }
+
+  function validate(event) {
+    assertHas(event.name, 'Please provide the name of the event')
+    assertHas(event.location, 'Please provide a location')
+    assertHas(event.start_date, 'Please tell us the start date')
+    assertHas(event.end_date, 'Please tell us the end date')
+  }
+
+  async function onSubmit() {
+    validate(event)
+
+    if (isNew) await create(event)
+    else await update(event)
+
+    $goto(`/events`)
+
+  }
+
+  function imageUploaded(e) {
+    event.image_file = e.detail
+  }
+
+  function onLocationChanged(e) {
+    event.location = e.detail
+  }
+
+</script>
+
+<style>
+  .preview {
+    max-height: 160px;
+  }
+</style>
+  
+{#if $events.length != 0 && !existingEvent && !isNew }
+  <h2 class="mb-e">Event does not exist!</h2>photo
+{:else if existingEvent && event.created_by?.id != $me.id && !isNew }
```

Comment:
```suggestion
{:else if !isNew && !existingEvent?.is_editable }
```
`is_editable` is a new property that should be used so that the backend is in control of authorization.